Sources/DistributedActors/ClusterSystem.swift

```diff
@@ -1493,12 +1493,26 @@ public struct ClusterInvocationResultHandler: DistributedTargetInvocationResultH
             system.log.debug("Result handler, onThrow: \(error)", metadata: ["call/id": "\(callID)"])
             let reply: RemoteCallReply<_Done>
             if let codableError = error as? (Error & Codable) {
-                reply = .init(callID: callID, error: codableError)
+                switch system.settings.remoteCall.codableErrorAllowance {
+                case .custom(let allowedTypes) where errorTypeAllowed(error: error, allowedTypes: allowedTypes):
+                    reply = .init(callID: callID, error: codableError)
+                case .all: // compiler gets confused if this is grouped together with above
+                    reply = .init(callID: callID, error: codableError)
+                default:
+                    reply = .init(callID: callID, error: GenericRemoteCallError(message: "Remote call error of [\(type(of: error as Any))] type occurred"))
+                }
             } else {
                 reply = .init(callID: callID, error: GenericRemoteCallError(message: "Remote call error of [\(type(of: error as Any))] type occurred"))
             }
             try await channel.writeAndFlush(TransportEnvelope(envelope: Payload(payload: .message(reply)), recipient: recipient))
         }
+
+        func errorTypeAllowed<Err>(error: Err, allowedTypes: [(Error & Codable).Type]) -> Bool {
+            func isErrorOfType<Err, T: Error & Codable>(_ error: Err, _: T.Type) -> Bool {
+                error is T // `is` and `as` don't work with stored type
+            }
+            return allowedTypes.contains { type in isErrorOfType(error, type) }
```

Comment:
Optimization idea, can be as follow up:

We should be able to form a `Set<ObjectIdentifier>` for the types that are allowed via `ObjectIdentifier(the type)` and store this in settings when users give us their `Some.self`.

And then thanks to object equality of the identifiers (`ObjectIdentifier(String.self) == (ObjectIdentifier(String.self as any Any.Type))`) we should be able to make this check a `allowedTypeObjectIDs.contains(ObjectIdentifier(Err.self))`

There may be some type trickery... so if it does not "just™" work please make a ticket and we'll follow up on it :-)

Reply:
I have to do `type(of: error as Any)` to get the right type, otherwise `Err` is just `Error`.

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/1020/commits/046cd443dfd0542aa04117d368e480afe6a49c3b